File: src/main/java/org/cloudfoundry/credhub/view/FindCertificateResult.java
package org.cloudfoundry.credhub.view;

import java.time.Instant;

import com.fasterxml.jackson.annotation.JsonProperty;

public class FindCertificateResult extends FindCredentialResult {
  private final Instant expiryDate;

  public FindCertificateResult(final Instant versionCreatedAt, final String name, final Instant expiryDate) {
    super(versionCreatedAt, name);
    this.expiryDate = expiryDate;
  }

  @JsonProperty
  public Instant getExpiryDate() {
    return expiryDate;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/FindCertificateResult.kt
package org.cloudfoundry.credhub.views

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant

class FindCertificateResult(versionCreatedAt: Instant, name: String, @get:JsonProperty val expiryDate: Instant) : FindCredentialResult(versionCreatedAt, name)
